docker/Dockerfile.musica

```diff
@@ -0,0 +1,119 @@
+# parts of CAM require x86 architecture (gptl, which relies on the rdtsc x86 assembly instruction)
+# esmf is am image you are expected to have built. Read the README file for instructions
+FROM esmf:latest
+
+###################################################
+## Install necessary packages
+###################################################
+RUN dnf -y update \
+    && dnf -y install \
+      ftp \
+      git \
+      hostname \
+      m4 \
+      python \
```

Comment:
Similar to the python comment above, would it be worthwhile to update this to `python3` to be explicit?  I think most linux distros just alias this at install time to check for `python3` so it might not be necessary.

Reply:
This does install python3 by default. I'm good to be explicit if you prefer though

Reply:
If it's not too much trouble, I'd be happy to push the change directly to the branch.

Reply:
Fine with me!

Reply:
So when changing this to `python3`, the build produces the log output: `Package python3-3.11.4-1.fc38.x86_64 is already installed.`  I originally thought this was fine but the build ultimately failed with:
```
STEP 18/37: RUN ./manage_externals/checkout_externals
/usr/bin/env: 'python': No such file or directory
Error: building at STEP "RUN ./manage_externals/checkout_externals": while running runtime: exit status 127
```

Leaving this as `python` is actually installing the package `python-unversioned-command`:
```
STEP 2/37: RUN dnf -y update     && dnf -y install       ftp       git       hostname       m4       python       sudo       svn       tree       vim     && dnf clean all
Fedora 38 - x86_64                               32 MB/s |  83 MB     00:02
Fedora 38 openh264 (From Cisco) - x86_64        2.6 kB/s | 2.5 kB     00:00
Fedora Modular 38 - x86_64                      935 kB/s | 2.8 MB     00:03
Fedora 38 - x86_64 - Updates                     12 MB/s |  30 MB     00:02
Fedora Modular 38 - x86_64 - Updates            2.0 MB/s | 2.1 MB     00:01
Dependencies resolved.
Nothing to do.
Complete!
Last metadata expiration check: 0:00:02 ago on Wed Aug 16 15:16:56 2023.
Package sudo-1.9.13-2.p2.fc38.x86_64 is already installed.
Dependencies resolved.
================================================================================
 Package                     Arch    Version                     Repo      Size
================================================================================
Installing:
 ftp                         x86_64  0.17-91.fc38                fedora    62 k
 git                         x86_64  2.41.0-1.fc38               updates   54 k
 hostname                    x86_64  3.23-8.fc38                 fedora    27 k
 m4                          x86_64  1.4.19-5.fc38               fedora   303 k
 python-unversioned-command  noarch  3.11.4-1.fc38               updates   11 k
 ...
```
which is what causes the build to succeed.
 
Given that this requires a change to the `checkout_externals` script, I'm okay leaving this as is instead.